### PR TITLE
[Refactor] Write a smoke test more easily

### PR DESCRIPTION
**Pull request by ybiquitous**
This change aims to make it easier writing a smoke test by improving the `Smoke.add_test` API.
The current API has a problem that it can produce many duplications.

I recommend [viewing the diff with ignored whitespaces](https://github.com/sider/runners/pull/989/files?diff=unified&w=1).

### CODE REVIEW

**Comment by ybiquitous** [comment]
@yykamei What do you think? Can I proceed?

**Comment by yykamei** [comment]
Yes, you can. Looks useful.

**Comment by ybiquitous** [comment]
OK, thanks. 👍

File: test/smokes/brakeman/expectations.rb [hunk 1]
```diff
@@ -1,180 +1,160 @@
-Smoke = Runners::Testing::Smoke
+s = Runners::Testing::Smoke
```
Comment (ybiquitous) [inline test/smokes/brakeman/expectations.rb:1]: [note] `Smoke` is a constant, that is we should define a type signature for it. So, I change it to a local variable.

**Comment by ybiquitous** [comment]
Oh, `rbprettier` formats and brakes syntax... 😢

**Comment by ybiquitous** [comment]
> Oh, `rbprettier` formats automatically on pre-commit and brakes syntax... 😢

https://github.com/prettier/plugin-ruby/issues/316

File: lib/runners/testing/smoke.rb [hunk 2]
```diff
@@ -127,22 +131,28 @@ def self.only?(name)
         end
       end
 
-      def self.add_test(name, result, **others)
+      def self.add_test(name, type:, guid: "test-guid", timestamp: :_,
```
Comment (ybiquitous) [inline lib/runners/testing/smoke.rb:134]: > __[RuboCop]__ Avoid parameter lists longer than 7 parameters. [13/7] ([view](https://sider.review/gh/repos/202486306/pulls/989#issue-5497595))
>
> <details>
>
> | Rule | Severity |
> | - | - |
> | `Metrics/ParameterLists` | convention |
>
> References:
> - <https://rubystyle.guide#too-many-params>
> - <https://github.com/rubocop-hq/rubocop/blob/v0.80.1/manual/cops_metrics.md#metricsparameterlists>
>
> *You can [mark this issue as **closed**](https://sider.review/gh/repos/202486306/pulls/989#issue-5497595) if you should not tackle it. (see our [document](https://help.sider.review/getting-started/working-with-issues) about issues)*
>
> </details>

Yes, long. But necessary.

File: test/smokes/fxcop/expectations.rb [hunk 3]
```diff
@@ -1,201 +1,173 @@
...
-        links: ["https://docs.microsoft.com/visualstudio/code-quality/ca1801-review-unused-parameters"]
-      }
-    ],
-    analyzer: { name: 'FxCop', version: '2.9.8' }
-  }
+      git_blame_info: nil,
+      links: ["https://docs.microsoft.com/visualstudio/code-quality/ca1801-review-unused-parameters"]
+    }
+  ],
+  analyzer: { name: 'FxCop', version: '2.9.8' }
```
Comment (ybiquitous) [inline test/smokes/fxcop/expectations.rb:172]: [note] To minimize conflicts with PR #971, this file is NOT auto-fixed by Prettier.

**Comment by ybiquitous** [comment]
@yykamei I'm sorry for the big diffs, but could you please review this if you have time?

**Comment by ybiquitous** [comment]
Thank you for your review! 🙇

**Pull request merged**